treehouses/cli
# `treehouses shadowsocks` (fixes #1596)

fixes #1596 

## how to test
The easiest way to see if it's working is to see if your ip address changes.

**Be sure to use [this image](http://download.treehouses.io/experiment/treehouse-shadowsocks1.img.gz) for testing, or manually install the two dependent packages** 
`apt install -y shadowsocks-libev  proxychains4`

You can use this json file for testing
```
{
                "server": "sgp02.clashcloud.tech",
                "local_address": "127.0.0.1",
                "local_port": 1080,
                "timeout": 300,
                "server_port": 58817,
                "password": "1vgfdr",
                "method": "rc4-md5"
}
```
- do `./cli.sh shadowsocks add` then give a **name** for this config

- paste the json file inside, save and quit

- do `./cli.sh shadowsocks start [name of the config]`

- upon success, do `./cli.sh shadowsocks enter [name of the config]`

now you are in a new shell session proxied by shadowsocks client!

do `curl ipinfo.io` to see if your ip address and location changes.

---

After adding more to it, a sample output is obtained:
```
root@treehouses:/cli# ./cli.sh shadowsocks list
Config:

hk
TW
us2
us

Running:

CONFIG          PORT                    LOCATION
hk              1081             HK -  Central and Western
us2             1082             US -  California
TW              1083             TW -  Taiwan

Failed:

us
        Run `journalctl -u shadowsocks-libev-local@us.service` for more info.
```
